From 32e8cffbcc3b0c6254ce75dbebe4278959a165bf Mon Sep 17 00:00:00 2001
From: Rui Wang <ruwang@redhat.com>
Date: Thu, 21 Aug 2014 04:13:50 -0400
Subject: [edac] sb_edac: Modify H/W event reporting policy

Message-id: <1408594432-28006-16-git-send-email-ruwang@redhat.com>
Patchwork-id: 88491
O-Subject: [PATCH RHEL7.1 BZ1076655 15/17] EDAC, sb_edac: Modify H/W event reporting policy
Bugzilla: 1076655
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Chen, Gong" <gong.chen@linux.intel.com>

Backport of:
Original commit: fd521039666529a4674b9822f1cc873672f57ee6
Original Author: "Chen, Gong" <gong.chen@linux.intel.com>
Original Date: Fri, 6 Dec 2013 01:17:09 -0500

 Newer Intel platforms support more than one method to report H/W event.
 On this kind of platform, H/W event report can adopt new method and
 traditional EDAC method should be disabled. Moreover, if EDAC event
 report method is set to *force*, it means event must be reported via
 EDAC interface. IOW, it overrides the default event report policy.

Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/sb_edac.c b/drivers/edac/sb_edac.c
index 80f9974..e06dc40 100644
--- a/drivers/edac/sb_edac.c
+++ b/drivers/edac/sb_edac.c
@@ -2071,6 +2071,9 @@ static int sbridge_mce_check_error(struct notifier_block *nb, unsigned long val,
  struct sbridge_pvt *pvt;
  char *type;
 
+ if (get_edac_report_status() == EDAC_REPORTING_DISABLED)
+  return NOTIFY_DONE;
+
  mci = get_mci_for_node_id(mce->socketid);
  if (!mci)
   return NOTIFY_BAD;
@@ -2421,9 +2424,10 @@ static int __init sbridge_init(void)
  opstate_init();
 
  pci_rc = pci_register_driver(&sbridge_driver);
-
  if (pci_rc >= 0) {
   mce_register_decode_chain(&sbridge_mce_dec);
+  if (get_edac_report_status() == EDAC_REPORTING_DISABLED)
+   sbridge_printk(KERN_WARNING, "Loading driver, error reporting disabled.\n");
   return 0;
  }
 
-- 
1.7.1